server/src/main/java/org/candlepin/model/PoolCurator.java

```diff
@@ -812,6 +813,22 @@ public boolean exists(Pool pool) {
                 .getSingleResult() > 0;
     }
 
+    /**
+     * Uses a database query to list pool ids of pools that require the given host for the given owner
+     * @param host the host consumer that is required
+     * @param owner the owner of the desired pools
+     * @return A list of pool ids that require the given host owned by the given owner
+     */
+    public List<String> getPoolsProvidedByHost(Consumer host, Owner owner) {
```

Comment:
do we need the owner as a separate argument? can we get that information from the consumer?